src/float.rkt

```diff
@@ -71,8 +50,111 @@
 (define (bit-difference x y)
   (ulps->bits (+ 1 (abs (ulp-difference x y)))))
 
-(define (sample-float)
-  (floating-point-bytes->real (integer->integer-bytes (random-exp 32) 4 #f)))
+(define (ordinary-value? x)
+  (match x
+    [(? real?)
+     (not (or (infinite? x) (nan? x)))]
+    [(? complex?)
+     (and (ordinary-value? (real-part x)) (ordinary-value? (imag-part x)))]
+    [(? boolean?)
+     true]
+    [(? posit8?)
+     (not (set-member? (representation-special-values posit8) x))]
+    [(? posit16?)
+     (not (set-member? (representation-special-values posit16) x))]
+    [(? posit32?)
+     (not (set-member? (representation-special-values posit32) x))]
+    [_ true]))
```

Comment:
Any reason this isn't using `representation-special-values` for the real, complex, and boolean cases?

Reply:
Good point, fixed so we use `representation-special-values` for real and bool cases, but left it off for complex. Right now there isn't any good way of handling complex since it's a different type (bool is too, but it fits a lot more cleanly into the interface without requiring major thought throughout the rest of the code.) This will be cleaned up in a future phase when we split off complex into a more clearly-defined type rather than the pseudo-representation that it is now.